cumulus/client/cirrus-executor/src/bundle_producer.rs

```diff
@@ -264,10 +270,44 @@ where
                 );
 
                 if u128::from(election_solution) <= threshold {
-                    // TODO: Add ProofOfElection into bundle so that farmer can verify the bundle
-                    // election.
+                    let storage_keys = well_known_keys::bundle_election_storage_keys();
+                    let storage_proof = self.client.read_proof(
+                        &best_block_id,
+                        &mut storage_keys.iter().map(|s| s.as_slice()),
+                    )?;
```

Comment:
Hm... looks like this "proof" actually contains all the data within it. Isn't it a problem that such "proof" is going to be huge? I was expecting that there would be proofs that certain values are within state root instead, like 3 Merkle Tree proofs and all the inputs to those, rather than the full maps/vectors.

Reply:
Make sense, but not feasible this time as the entire authoritiy set is stored as a single storage value in the runtime, let's use a TODO here and think about it on the next iteration, will make it a relative priority on my list.